domain/transfer.go

```diff
@@ -171,3 +180,42 @@ func (TransferProcessHandler) HandleEvent(
 
 	return nil
 }
+
+// HandleTimeout handles a timeout message that has been routed to this handler.
+func (TransferProcessHandler) HandleTimeout(
+	ctx context.Context,
+	s dogma.ProcessTimeoutScope,
+	m dogma.Message,
+) error {
+	switch x := m.(type) {
+	case TransferAtScheduledTimeout:
+		s.ExecuteCommand(commands.DebitAccount{
+			TransactionID:   x.TransactionID,
+			AccountID:       x.FromAccountID,
+			TransactionType: messages.Transfer,
+			Amount:          x.Amount,
+			ScheduledDate:   x.ScheduledDate,
+		})
+
+	default:
+		panic(dogma.UnexpectedMessage)
+	}
+
+	return nil
+}
+
+// TimeoutHint returns a duration that is suitable for computing a deadline for
+// the handling of the given message by this handler.
+func (TransferProcessHandler) TimeoutHint(m dogma.Message) time.Duration {
+	return 0
+}
+
+// TransferAtScheduledTimeout is a timeout command requesting that funds be
+// transferred from one bank account to another.
+type TransferAtScheduledTimeout struct {
```

Comment:
The more I see this the more I think we need to come up with some naming guidelines for timeout messages, like we have for commands and events.

I kind of like the sound of `TransferReadyTo<blah>`, for example.  It's a statement of fact, but not past-tense like  events are named, and it's not an imperative, the way a command is named.

What do you think?

Reply:
I renamed this timeout message to `TransferReadyToSend`. Does that match your plan for naming?

Reply:
I like it generally, but I'm hunting for a "better" word than "send" (one that's in the existing ubiquitous language, if possible.

I'm still happy with the kind of "present-tense" naming as a guideline, how about you?